.circleci/config.yml

```diff
@@ -1,104 +1,267 @@
+                  command: sudo npm i -g gulp-cli
+            - run:
+                  name: Install Node Packages
+                  command: yarn
+            - save_cache:
+                  name: Save Yarn Package Cache
+                  key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
+                  paths:
+                      - node_modules/
+            - run:
+                  name: Set Sentry
+                  command: echo {\"dsn\":\"${SENTRY_DSN}\"} > src/main/assets/data/sentry.json
+            - run:
+                  name: Build App
+                  command: yarn dist --publish=never
+            - persist_to_workspace:
+                  root: .
+                  paths:
+                      - dist
+            - store_artifacts:
+                  path: dist
+    'e2e-mac':
+        macos:
+            xcode: '9.0'
+        steps:
+            - checkout
+            - attach_workspace:
+                  at: /tmp/mac
+            - restore_cache:
+                  name: Restore Yarn Package Cache
+                  keys:
+                      - yarn-packages-mac-{{ .Branch }}-{{ checksum "yarn.lock" }}
+                      - yarn-packages-mac-{{ .Branch }}
+                      - yarn-packages-mac-master
+                      - yarn-packages-mac-
+            - run:
+                  name: Upgrade Node
+                  command: brew upgrade node
+            - run:
+                  name: Upgrade Yarn
+                  command: brew upgrade yarn
+            - run:
+                  name: Add Read Access to yarn config
+                  command: |
+                      sudo mkdir ~/.config
+                      sudo chown -R $USER:$GROUP ~/.config
+            - run:
+                  name: E2E Tests
+                  command: OSENV=circle-mac yarn test:e2e
 
-workflows:
-  version: 2
-  workflow1:
-    jobs:
-      - "build"
- #     - "e2e":
- #         requires:
- #           - build
-      - "stage":
-          requires:
-            - build
+    'stage-mac':
+        docker: *DOCKERIMAGE
+        steps:
+            - checkout
+            - attach_workspace:
+                  at: /tmp/mac
+            - run:
+                  name: Install gcloud
+                  command: |
+                      export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)"
+                      echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+                      curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+                      sudo apt-get update && sudo apt-get install google-cloud-sdk jq
+            - run:
+                  name: Deploy mac to staging
+                  command: bash deploy.sh "$(git log --format=%B -n 1 $CIRCLE_SHA1)"
+                  environment:
+                      OSENV: mac
 
+    'stage-linux':
+        docker: *DOCKERIMAGE
```

Comment:
Since we're using the docker executor here, we could just use the Cloud SDK docker image (https://hub.docker.com/r/google/cloud-sdk/) here instead of manually installing it.